src/argilla_sdk/records/_helpers.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2024-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict
+
+
+def _dict_to_record(record_dict: Dict[str, Any]) -> "Record":
```

Comment:
This cannot be part of the `Record.from_dict` because we're expecting the same record data structure as the `Record` class, right?

Reply:
We have the "reverse" action in the `GenericExportMixin._record_to_dict` class. Maybe we could move this code there since the logic are quite related?

Reply:
> We have the "reverse" action in the `GenericExportMixin._record_to_dict` class. Maybe we could move this code there since the logic is quite related?

This was my first implementation, but it became complex with the iterator. I can refactor into though.

Reply:
> This cannot be part of the `Record.from_dict` because we're expecting the same record data structure as the `Record` class, right?

The dict structure is actually different to `from_dict`, because it deals with dot notation etc.

Reply:
> The dict structure is actually different to from_dict, because it deals with dot notation etc.

Yes, I realized this later. 

We have `Record.from_dict`, `Record.to_dict` and the export utilities. `to_dict` is related to the export (used internally) but, regarding the class definition, I would expect to do `Record.from_dict(record.to_dict))` (Not saying we should implement it)

Reply:
I agree that this is strang and `Record.from_dict(record.to_dict))` would be expected. Do you think some of this routes should be made private?

Reply:
Sorry for missing this. I will come back here after the revision

Reply:
I think we can take a look to the `Record.from_dict` method and support also this conversion. But this would be tackled in a separate PR. 

As a user, I would expect to create a record from a serialization:
```python

json_data = ...  # read some records file as a json list

record = Record.from_dict(json_data[0])
```

But also (or not), 
```python
record = Record.from_dict({ "text": ..., "label": ....}, mapping= ....)
```

We have (at least) two alternatives:

1. Add support to `Record.from_dict` method to create records from a record serialization (to_dict())
2. Align `Record.from_dict` and `record.to_dict` methods and create a separate method to build records using mappings (with a default mapping defined by the dataset schema).

I'm not sure what should be the best option. I have a feeling about the second one since the from_dict + mapping is used only internally.

Reply:
In general, I agree that point 2 is more ideal. But I think that this should be tackled as part of a refactor of the ingestion and `Record.from_dict`.

That said, I'd like to get some wider insights on this flow from users in relation to the add/update/log/upsert discussion around `DatasetRecords`.

In the meantime, how would feel about making some of these methods private? So that they don't confuse the core workflow.

Reply:
> In the meantime, how would feel about making some of these methods private? So that they don't confuse the core workflow.

It sounds great! But feel free to move these changes into a separate PR or here (the best match for you)